### PR TITLE
OpenDream Cleanup Pass

### PR DESCRIPTION
**Pull request by ike709**
## About The Pull Request

OpenDream can detect BYONDisms that BYOND or SpacemanDMM miss. This PR fixes all of the issues it found in TG. Each change is explained in code comments below.

Behavior is identical.

## Why It's Good For The Game

BYONDisms bad.

## Changelog

Nothing player-facing.

### CODE REVIEW

File: code/datums/components/construction.dm [hunk 1]
```diff
@@ -91,8 +91,9 @@
 				. = user.transferItemToLoc(I, parent)
 
 			// Using stacks
-			else if(istype(I, /obj/item/stack))
-				. = I.use_tool(parent, user, 0, volume=50, amount=current_step["amount"])
+			else
+				if(istype(I, /obj/item/stack))
+					. = I.use_tool(parent, user, 0, volume=50, amount=current_step["amount"])
```
Comment (ike709) [inline code/datums/components/construction.dm:96]: `else if()` is not a valid switch case nor is it evaluated as such. In this case there is no undesirable behavior, but it's still better to fix it.

File: code/datums/components/construction.dm [hunk 2]
```diff
@@ -108,8 +109,9 @@
 				if(located_item)
 					located_item.forceMove(drop_location())
 
-			else if(ispath(target_step_key, /obj/item/stack))
-				new target_step_key(drop_location(), target_step["amount"])
+			else
+				if(ispath(target_step_key, /obj/item/stack))
+					new target_step_key(drop_location(), target_step["amount"])
```
Comment (ike709) [inline code/datums/components/construction.dm:114]: Ditto.

File: code/modules/cargo/supplypod.dm [hunk 3]
```diff
@@ -598,7 +598,7 @@
 	if(pod.effectStun) //If effectStun is true, stun any mobs caught on this pod_landingzone until the pod gets a chance to hit them
 		for (var/mob/living/target_living in get_turf(src))
 			target_living.Stun(pod.delays[POD_TRANSIT]+10, ignore_canstun = TRUE)//you ain't goin nowhere, kid.
-	if (pod.delays[POD_FALLING] == initial(pod.delays[POD_FALLING]) && pod.delays[POD_TRANSIT] + pod.delays[POD_FALLING] < pod.fallingSoundLength)
+	if (pod.delays[POD_TRANSIT] + pod.delays[POD_FALLING] < pod.fallingSoundLength)
```
Comment (ike709) [inline code/modules/cargo/supplypod.dm:601]: `initial()` on a list index (except for the `vars` list) returns the current value. This is equivalent to `thing == thing`.

File: code/modules/mafia/roles.dm [hunk 4]
```diff
@@ -467,7 +467,6 @@
 /datum/mafia_role/lawyer/proc/release(datum/mafia_controller/game)
 	SIGNAL_HANDLER
 
-	. = ..()
```
Comment (ike709) [inline code/modules/mafia/roles.dm:470]: Parent call in proc definition. You don't want to know when that has actual effects, but this isn't one of those cases. This calls nothing.

File: code/modules/modular_computers/file_system/programs/ntmessenger.dm [hunk 5]
```diff
@@ -299,7 +299,7 @@
 			to_chat(M, "[FOLLOW_LINK(M, user)] [ghost_message]")
 
 	// Log in the talk log
-	user.log_talk(message, LOG_PDA, tag="[rigged ? "Rigged" : ""] PDA: [initial(message_data["name"])] to [signal.format_target()]")
+	user.log_talk(message, LOG_PDA, tag="[rigged ? "Rigged" : ""] PDA: [message_data["name"]] to [signal.format_target()]")
```
Comment (ike709) [inline code/modules/modular_computers/file_system/programs/ntmessenger.dm:302]: Ditto `initial()` on list index.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 6]
```diff
@@ -60,8 +60,9 @@
 				mytray.mutateweed()
 			if(1   to 32)
 				mytray.mutatepest(user)
-			else if(prob(20))
-				mytray.visible_message(span_warning("Nothing happens..."))
+			else
+				if(prob(20))
+					mytray.visible_message(span_warning("Nothing happens..."))
```
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:65]: Ditto switch case.

File: code/modules/cargo/supplypod.dm [hunk 7]
```diff
@@ -598,7 +598,7 @@
 	if(pod.effectStun) //If effectStun is true, stun any mobs caught on this pod_landingzone until the pod gets a chance to hit them
 		for (var/mob/living/target_living in get_turf(src))
 			target_living.Stun(pod.delays[POD_TRANSIT]+10, ignore_canstun = TRUE)//you ain't goin nowhere, kid.
-	if (pod.delays[POD_FALLING] == initial(pod.delays[POD_FALLING]) && pod.delays[POD_TRANSIT] + pod.delays[POD_FALLING] < pod.fallingSoundLength)
```
Comment (SuperNovaa41) [inline code/modules/cargo/supplypod.dm:601]: are you sure this doesn't change anything?

File: code/modules/cargo/supplypod.dm (hunk 3 above)
Comment (SuperNovaa41) [reply code/modules/cargo/supplypod.dm:601]: hilarious

**Comment by ike709** [comment]
@SuperNovaa41 I saw the comment you deleted so here's proof:
image

File: code/modules/mafia/roles.dm (hunk 4 above)
Comment (Mothblocks) [reply code/modules/mafia/roles.dm:470]: I want to know when this has actual effects 🥺

**Review by Kylerace — APPROVED** [review]

File: code/modules/mafia/roles.dm (hunk 4 above)
Comment (ike709) [reply code/modules/mafia/roles.dm:470]: Trust me, you don't and you'll never encounter it unless I open that Pandora's Box for you.

I'll give you one truly cursed hint: `_dm_interface = 129`

**Comment by SuperNovaa41** [comment]
> @SuperNovaa41 I saw the comment you deleted so here's proof: image

I saw your review after i posted that :D

File: code/modules/food_and_drinks/restaurant/custom_order.dm [hunk 8]
```diff
@@ -35,11 +35,10 @@
 /datum/custom_order/moth_clothing/New(datum/venue/our_venue)
 	var/mob/living/carbon/buffet = our_venue.restaurant_portal?.turned_on_portal?.resolve()
 	if (!istype(buffet)) // Always asks for the clothes that you have on, but this is a fallback.
-		wanted_clothing_type = pick_weight(
+		wanted_clothing_type = pick_weight(list(
```
Comment (ike709) [inline code/modules/food_and_drinks/restaurant/custom_order.dm:38]: `pick_weight()` takes a list.

File: code/modules/mafia/roles.dm (hunk 4 above)
Comment (Mothblocks) [reply code/modules/mafia/roles.dm:470]: 🤐<-- PUN TO INDICATE I HAVE LEARNED WHAT IT IS

Comment (ike709) [reply code/modules/mafia/roles.dm:470]: MSO also reminded me that they're necessary in proc redefinitions at the same level, but that wasn't what I was referring to.

File: code/modules/cargo/supplypod.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/modules/cargo/supplypod.dm:601]: So the check for if we're using the same delays never worked?
Shouldn't a better check be added then?

Comment (ike709) [reply code/modules/cargo/supplypod.dm:601]: @LemonInTheDark Such as?

File: code/datums/components/squeak.dm [hunk 9]
```diff
@@ -130,11 +130,11 @@
 		holder = null
 
 // Disposal pipes related shits
-/datum/component/squeak/proc/disposing_react(datum/source, obj/structure/disposalholder/holder, obj/machinery/disposal/source)
+/datum/component/squeak/proc/disposing_react(datum/source, obj/structure/disposalholder/disposal_holder, obj/machinery/disposal/disposal_source)
```
Comment (ike709) [inline code/datums/components/squeak.dm:133]: BYOND doesn't error on duplicate args. Instead it just uses the first one. Renamed the other arg for consistency.

File: code/modules/food_and_drinks/restaurant/custom_order.dm [hunk 10]
```diff
@@ -35,11 +35,10 @@
...
 			/obj/item/clothing/head/chefhat = 3,
 			/obj/item/clothing/shoes/sneakers/black = 3,
-			/obj/item/clothing/gloves/color/black = 1,
-		)
+			/obj/item/clothing/gloves/color/black = 1))
```
Comment (Fikou) [inline code/modules/food_and_drinks/restaurant/custom_order.dm:41]:
```suggestion
			/obj/item/clothing/gloves/color/black = 1,
		))
```

File: code/modules/cargo/supplypod.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/modules/cargo/supplypod.dm:601]: In retrospect I'm not sure why this was even done

**Pull request merged**